cmd/juju/application/store/charmadapter_test.go

```diff
@@ -109,8 +110,8 @@ func (s *resolveSuite) TestResolveCharmFailResolveWithChannel(c *gc.C) {
 		return s.downloadClient, nil
 	})
 	_, obtainedOrigin, _, err := charmAdapter.ResolveCharm(curl, origin, false)
-	c.Assert(err, gc.NotNil)
-	c.Assert(obtainedOrigin.Risk, gc.Equals, string(csparams.NoChannel))
+	c.Assert(err, gc.IsNil)
```

Comment:
Do we then need to change the test name from `TestResolveCharmFailResolveWithChannel`? It's not a fail scenario now.

Reply:
So it is a ResolveCharm fail, but a success fallback scenario. I'll amend.